### PR TITLE
reduced a for loop and exposed a function 4 users.

### PR DESCRIPTION
**Pull request by ludlows**
1. The origin version of function _BalancingTransformer.transform_array_ contained a redundant _for loop_ to use self.weights for computing matrix  _w\_balanced_. This computation can be replaced by one _for loop_.

2. Given _y_ and _w_,  maybe some users only want to compute _w\_balanced_. I think it is better to expose this function to users. So I created static method  _compute\_balance\_weights_.

### CODE REVIEW

File: deepchem/trans/transformers.py [hunk 1]
```diff
@@ -354,33 +354,33 @@ def __init__(self,
     w = self.dataset.w
     # Ensure dataset is binary
     np.testing.assert_allclose(sorted(np.unique(y)), np.array([0., 1.]))
-    weights = []
-    for ind, task in enumerate(self.dataset.get_task_names()):
+    self.balance_w = self.compute_balance_weights(y, w)
+
+  @staticmethod
+  def compute_balance_weights(y, w):
```
Comment (rbharath) [inline deepchem/trans/transformers.py]: @ludlows Can you add a unit test for this function?

File: deepchem/trans/transformers.py [hunk 2]
```diff
@@ -354,33 +354,33 @@ def __init__(self,
...
+      # Find indices of inactive and active sample for each task
+      zero_indices = np.logical_and(task_y == 0, task_w != 0)
+      one_indices = np.logical_and(task_y == 1, task_w != 0)
+      balance_w[zero_indices, ind] = neg_weight
+      balance_w[one_indices, ind] = pos_weight
+    return balance_w
 
   def transform_array(self, X, y, w):
     """Transform the data in a set of (X, y, w) arrays."""
-    w_balanced = np.zeros_like(w)
```
Comment (rbharath) [inline deepchem/trans/transformers.py]: I don't believe this is correct. There's no guarantee that `self.balance_w` is of the same shape as `w`

**Comment by rbharath** [comment]
@ludlows Thanks for the PR. I'm not entirely convinced that this code is doing the right thing though. Could you add more unit tests? 

Also, it might help if you created a issue explaining what the bug in the current code is along with a failing test case.

File: deepchem/trans/transformers.py (hunk 2 above)
Comment (ludlows) [inline deepchem/trans/transformers.py]: Hi, Bharath.  I think the shape of w and self.balance_w should be the same one like (num_sample, num_task) . Could you explain why their shapes are not same? Thanks.

File: deepchem/trans/transformers.py (hunk 1 above)
Comment (ludlows) [inline deepchem/trans/transformers.py]: Writing it right now.

File: deepchem/trans/transformers.py (hunk 2 above)
Comment (rbharath) [inline deepchem/trans/transformers.py]: `transform_array` can be called on arrays of different shape than those used to construct the Transformer object. So there's no guarantee that the new `w` has the same shape as the one from the constructor.

**Pull request closed without merging**